nacos-spring-context/src/main/java/com/alibaba/nacos/spring/core/env/NacosPropertySourcePostProcessor.java

```diff
@@ -86,8 +88,13 @@ public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry) t
 
     @Override
     public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory) throws BeansException {
+        this.nacosPropertySourceBuilders = new ArrayList<AbstractNacosPropertySourceBuilder>(2);
```

Comment:
I suggest you get Bean names from `BeanUtils.getBeanNames(beanFactory,AbstractNacosPropertySourceBuilder.class)`, and then pass it's length as `ArrayList`'s capacity.